src/animal/animal.controller.ts

```diff
@@ -51,22 +62,22 @@ export class AnimalController {
   @ApiConsumes(APPLICATION_JSON_MEDIA_TYPE)
   @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
   @ApiCreatedResponse({ status: 201, description: 'Animal created', type: AnimalDto })
-  @ApiBadRequestResponse({
-    status: 400,
-    description: 'Request validation failed',
-  })
+  @ApiBadRequestResponse({ status: 400, description: 'Request validation failed' })
   @Post()
   @UsePipes(ValidationPipe)
-  async createAnimal(@Body() data: MutateAnimalDto): Promise<AnimalDto> {
+  async createAnimal(@Body() data: AnimalCreateDto): Promise<AnimalDto> {
     return await this.service.create(fromAnimalMutateDtoToModel(data)).then((value) => fromAnimalModelToDto(value));
   }
 
   @Put(':id')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
-  @ApiResponse({ description: 'Animal updated successfully.', type: AnimalDto })
+  @ApiConsumes(APPLICATION_JSON_MEDIA_TYPE)
```

Comment:
You assign `@ApiConsumes` and `@ApiProduces` to a lot of methods, please consider using [decorator composition](https://docs.nestjs.com/custom-decorators#decorator-composition) that apply these configurations.